### PR TITLE
[BENCHMARK WORKFLOW]

### PR DESCRIPTION
**Pull request by johncalesp**
- Added new yaml file for workflow
- Added new fields to argument, operator and predictor

### CODE REVIEW

File: analyzer/habitat/analysis/operation.py [hunk 1]
```diff
@@ -91,12 +92,18 @@ def __init__(
         measured_operation,
         forward,
         backward,
-        device
+        device,
+        measured_local=0,
```
Comment (jimgao1) [inline analyzer/habitat/analysis/operation.py:96]: * `measured_local` belongs in `self.measured_operation`. 
* `predicted_local` can be derived directly from `self.measured_operation` by calling `to_device` on it - no need to store a redundant copy.
* Similarly, `unscaled_predicted` can be derived from existing class members, so it need not be stored.

Hence we can eliminate all three additional init parameters, and instead include the implementation in subsequent getter functions.

File: analyzer/habitat/analysis/predictor.py [hunk 2]
```diff
@@ -140,7 +140,7 @@ def _wave_scale(self, run_time, dest_device):
         )
 
     def _special_scale(self, operation, dest_device, scaler):
-        predicted_ms = scaler(operation, dest_device)
+        predicted_ms, measured_local, predicted_local, unscaled_predicted = scaler(operation, dest_device)
```
Comment (jimgao1) [inline analyzer/habitat/analysis/predictor.py:143]: Can probably remove `measured_local` and `predicted_local`. 

We don't always need `unscaled_predicted`, so maybe include an optional parameter to `scaler` (e.g. `include_unscaled`) which enables returning this additional value, or just call it `scale=False` and call `scaler` twice. 

```python
predicted_ms = scaler(operation, dest_device) # existing API
unscaled_predicted = scaler(operation, dest_device, scale=False) # new additional option
```

File: analyzer/habitat/analysis/operation.py [hunk 3]
```diff
@@ -81,6 +81,7 @@ def device(self):
 
     def to_device(self, dest_device, predictor):
         if dest_device.name == self._device.name:
+            warnings.warn("Predicting to the same device")
             return self
```
Comment (jimgao1) [inline analyzer/habitat/analysis/operation.py:85]: Remember to delete this line so it falls through

**Review by jimgao1 — CHANGES_REQUESTED** [review]

File: analyzer/habitat/analysis/arguments.py [hunk 4]
```diff
@@ -5,9 +5,10 @@ class Arguments:
     """
     Stores representations of an operation's arguments.
     """
-    def __init__(self, args, kwargs):
+    def __init__(self, args, kwargs, debug_args):
```
Comment (jimgao1) [inline analyzer/habitat/analysis/arguments.py:8]: Add comment for why we need `debug_args`.

**Review by jimgao1 — CHANGES_REQUESTED** [review]

File: analyzer/habitat/analysis/arguments.py (hunk 4 above)
Comment (johncalesp) [reply analyzer/habitat/analysis/arguments.py:9]: added more comments

File: analyzer/habitat/analysis/operation.py (hunk 3 above)
Comment (johncalesp) [reply analyzer/habitat/analysis/operation.py]: done

File: analyzer/habitat/analysis/predictor.py (hunk 2 above)
Comment (johncalesp) [reply analyzer/habitat/analysis/predictor.py]: added optional parameter

**Review by jimgao1 — APPROVED** [review]

**Pull request merged**